### DISCUSSION TITLE
Kubuntu 23.04 and missing libqtchartsqml2plugin.so

### DISCUSSION THREAD

**Discussion by ilkant** [Stellarium/stellarium#3201] [General]
I just updated Kubuntu 22.10 -> 23.04. When I tried to compile Stellarium on my computer 

$ cmake -DCMAKE_INSTALL_PREFIX=/opt/stellarium ../..

CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Qt6Qml/QmlPlugins/Qt6qtchartsqml2Targets.cmake:96 (message):
  The imported target "Qt6::qtchartsqml2" references the file

     "/usr/lib/x86_64-linux-gnu/qt6/qml/QtCharts/libqtchartsqml2plugin.so"

  but this file does not exist.

That file is not in Kubuntu's apt repository too. With Kubuntu 22.10 compiling was succesfull.

**Comment by 10110111** [comment]
Debian provides this file in `qml6-module-qtcharts` package. If Ubuntu doesn't, it's a question to Ubuntu's packagers.

**Reply by ilkant** [reply]
Ubuntu repository has

http://fi.archive.ubuntu.com/ubuntu lunar/universe amd64 qml6-module-qtcharts amd64 6.4.2-2 [26,7 kB]

Now message is:

CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Qt6Qml/QmlPlugins/Qt6quick3dspatialaudioTargets.cmake:96 (message):
  The imported target "Qt6::quick3dspatialaudio" references the file

     "/usr/lib/x86_64-linux-gnu/qt6/qml/QtQuick3D/SpatialAudio/libquick3dspatialaudioplugin.so"

  but this file does not exist.

**Reply by 10110111** [reply]
Try using `apt-file` to find it.

**Reply by gzotti** [reply]
I wonder why it is even looking for anything QML.

**Reply by ilkant** [reply]
I removed Indi PPA and I remove packets from there. And perhaps whole KStars.

$ sudo add-apt-repository --remove ppa:mutlaqja/ppa

Now I have other activity and I continue with this issue some hours later. Thank you.

**Reply by alex-w** [reply]
Please check https://github.com/Stellarium/stellarium-ppa/blob/qt64/debian/control

**Comment by ilkant** [comment]
File CMakeCache.txt in stellarium/build/unix/ directory contains lines:

//Define whether the Scenery3d plugin should be created.
USE_PLUGIN_SCENERY3D:BOOL=1

Is this ok? Does that setting  cause Qt Qml errors? And which process produces CMakeCache.txt file?

**Reply by 10110111** [reply]
AFAIK, nothing in desktop Stellarium uses QML. And `CMakeCache.txt` is produced by `cmake` that you run before building.

**Reply by alex-w** [reply]
QML may be added as an dependency for QWidgets stuff

**Reply by gzotti** [reply]
Usually programs use either traditional widgets or more recent QML. Mixing is possible (see examples), but a dependency of the older system on the newer sounds very unlikely.

**Comment by gzotti** [comment]
You can try to deselect all optional parts, plugins (`USE_PLUGIN_*:BOOL=0`), Multimedia, XLS, ... to identify the critical component. However, given that it works on Kubuntu 22.10, and not on 23.04, this sounds like a Kubuntu 23.04 config/package issue that should be reported there.

**Comment by ilkant** [comment]
Thank you helping me! I found libquick3dspatialaudioplugin.so from Ubuntun package qml6-module-qtquick3d-spatialaudio. Then cmake worked without errors and compilation produced Stellarium verison 23.1.117-96f9d30 [master]. It's compiled today 27th April 2023.

Next task to me could be to see Stellarium code observe list exporting and importing. And I can try to solve that. But I am not very   competent of Stellarium code. :) I think that he who has coded Stellarium, does it much more efficiently and quickly.
